### ISSUE TITLE
Output version when running bin/plugin list

### ISSUE THREAD

**Issue by gmoskovicz** [elastic/elasticsearch#16375]
Currently, we only get the plugin list without versions:

```
Installed plugins in /Users/Gabriel/Documents/ElasticSearch/elasticsearch-2.1.1/plugins:
    - license
    - mapper-size
    - marvel-agent
    - watcher
```

It will be nice to print the version of each plugin so we can verify that the correct version is installed.

**Comment by dadoonet** [comment]
@gmoskovicz Note that for the examples you gave, version will be for all those plugins `2.1.1` as we release official plugins using elasticsearch version number.

**Comment by gmoskovicz** [comment]
@dadoonet Correct! However i think that for community plugins this can vary.

**Comment by electrical** [comment]
Would be nice to have since community plugins can have different versions.

**Comment by alehane** [comment]
@gmoskovicz many thanks for raising this, I'd like to see something along the lines of:

```
Installed plugins in /Users/Gabriel/Documents/ElasticSearch/elasticsearch-2.1.1/plugins:
    - license (2.0.0)
    - mapper-size (2.1.1)
    - marvel-agent (2.1.1)
    - watcher (2.0.1)
```

Note: versions are examples rather than real values :)

**Comment by gmoskovicz** [comment]
+1 for @electrical and @alehane

**Comment by dadoonet** [comment]
> However i think that for community plugins this can vary.

Sure. I was just talking about the examples you mentioned.

Note that the `_cat/plugins` API provides such information but you already know that it requires to have elasticsearch running and I guess here you need to have that information without elasticsearch running.

**Comment by gmoskovicz** [comment]
> Note that the _cat/plugins API provides such information but you already know that it requires to have elasticsearch running and I guess here you need to have that information without elasticsearch running.

Yep. The useful information from this is **before** starting Elasticsearch, so you can play a little bit with versions and so on. I believe that this will be useful for automated releases and version upgrade.

**Comment by rjernst** [comment]
What would you have the format look like? Should it be only when using "verbose" mode, kind of like `ls` vs `ls -l`? It seems like it should either be just the name, or the name and all the info from the plugin properties file?

**Comment by dadoonet** [comment]
+1 for verbose mode
Not sure if we should print all properties though.

**Comment by rjernst** [comment]
Well, that is what verbose means. :)

**Comment by rjernst** [comment]
I would at least expect the description with it? And java version? So at that point, why not all the info we have about the plugin.

**Comment by dadoonet** [comment]
Indeed.

**Comment by gmoskovicz** [comment]
+1 for verbose mode!

**Comment by marek-obuchowicz** [comment]
We are at the moment parsing `plugin-descriptor.properties` file as a workaround, but it seems very logic and natural to be able to get this information from `plugin` tool. In the era of API's and configuration management systems, it would be very helpful to be able to get this information in "straight" way. Lots of third-party plugins are built for very specific ES versions, so we should be able to manage it in easiest possible way.

**Comment by damianpfister** [comment]
+1 for more info on plugins without the need to start Elasticsearch

**Comment by gmoskovicz** [comment]
@rjernst @dadoonet looks like the [ListPluginsCommand](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/plugins/ListPluginsCommand.java) class is the one that is returning this information.

Currently is just looking at the directory and printing the name of the folder (which is the plugin name). With verbosity maybe we can print the full file name and read the `.properties` file and print the `description` and `version` ?

**Comment by AttilaForgacs** [comment]
+1 this comes handy

**Comment by gmoskovicz** [comment]
@AttilaForgacs @marek-obuchowicz @dadoonet @damianpfister please add your `+1` as the 👍  in the first comment so we can then get some noise :)

**Comment by gmoskovicz** [comment]
Closed by #18051
